crates/rpc-client/src/pathfinder/proofs.rs

```diff
@@ -114,12 +117,13 @@ pub fn verify_proof<H: HashFunctionType>(
     // The tree height is 251, so the first 5 bits are ignored.
     let start = 5;
     let mut index = start;
+    let mut non_existence_proof = false;
 
     loop {
         match trie_node_iter.next() {
             None => {
-                if index - start != DEFAULT_STORAGE_TREE_HEIGHT {
-                    return Err(ProofVerificationError::KeyNotInProof {
+                if non_existence_proof || (index - start != DEFAULT_STORAGE_TREE_HEIGHT) {
```

Comment:
The `non_existence_proof` bool doesn't seem necessary here, can't you return the `NonExistenceProof` when `non_existence_proof` is set to true below?